packages/testing/examples/useBlockMeta.test.tsx

```diff
@@ -1,10 +1,25 @@
 import { expect } from 'chai'
-import { useBlockMeta } from '@usedapp/core'
-import { renderWeb3Hook, sleep } from '../src'
-
+import { useBlockMeta, BlockNumberProvider, ChainStateProvider, MultiCall } from '@usedapp/core'
+import { deployMulticall, MockConnector, renderWeb3Hook, renderWeb3HookOptions, sleep } from '../src'
+import { MockProvider } from 'ethereum-waffle'
+import React from 'react'
 describe('useBlockMeta', () => {
+  let webHookOptions: renderWeb3HookOptions<{ children: React.ReactNode }>
+
+  beforeEach(async () => {
+    const mockProvider = new MockProvider()
+    const mockConnector = new MockConnector(mockProvider)
+    const multicallAddresses = await deployMulticall(mockProvider, mockConnector, MultiCall)
+    const wrapper: React.FC = ({ children }) => (
+      <BlockNumberProvider>
+        <ChainStateProvider multicallAddresses={multicallAddresses}>{children}</ChainStateProvider>
+      </BlockNumberProvider>
+    )
+    webHookOptions = { mockProvider, mockConnector, renderHook: { wrapper } }
+  })
+
   it('retrieves block timestamp and difficulty', async () => {
-    const { result, waitForCurrent } = await renderWeb3Hook(useBlockMeta)
+    const { result, waitForCurrent } = await renderWeb3Hook(useBlockMeta, webHookOptions)
```

Comment:
I think tests should be providing wrappers, i know its a major change in functionality but in my opinion it will make testing better in the future.

Just a opinion though.

Reply:
Sure, but you could add custom wrappers before.  Now you just need to add them by yourself in every single test. And the point of the testing library is to reduce that boilerplate.

Reply:
I think all the tests should be in one place so development in the future will be easier and less confusing.

Reply:
So other option is to move all the tests to something like /packages/testing/testCore

Reply:
Also MockBlockNumberProvider could be created as it's a simple provider also creation of mock provider,connector and multicall addresses could be moved to testing library so there is still some place for refactor.

I could commit some refactoring before this pull request is merged, but i think it's a good start